lib/rest/Twilio.js

```diff
@@ -249,18 +253,17 @@ Twilio.prototype.request = function request(opts) {
     headers.Accept = 'application/json';
   }
 
-  var uri = opts.uri;
-  if (this.region) {
-    var parts = _.split(uri, '.');
-
-    if (parts.length > 1 && !_.isEqual(parts[1], this.region)) {
-      uri = _.join(_.concat([parts[0], this.region], _.slice(parts, 1)), '.');
-    }
+  var uri = new url.URL(opts.uri);
+  var [product, subdomain, domain] = uri.hostname.split('.');
```

Comment:
I _think_ it's possible the uri is already going to contain an edge/region. Think pagination/links.